src/bot.coffee

```diff
@@ -171,30 +171,38 @@ class SlackBot extends Adapter
   # @private
   ###
   eventHandler: (event) =>
-    {user, channel, bot} = event
+    {user, channel} = event
 
     # Ignore anything we sent
     # NOTE: coupled to getting `rtm.start` data
-    return if (user && (user.id is @self.id || user.id is @self.bot_id)) || (bot && (bot.id is @self.bot_id))
+    return if user && (user?.id is @self.id)
+
+    
+    ###*
+    # Hubot user object in Brain.
+    # User can represent a Slack user OR bot
+    # 
+    # The returned user from a message or reaction event is guaranteed to contain:
+    # 
+    # id {String}:              Slack user ID
+    # slack.is_bot {Boolean}:   Flag indicating whether user is a bot
+    # name {String}:            Slack username
+    # real_name {String}:       Name of Slack user or bot
+    ###
+    user = if user?.id? then @robot.brain.userForId user.id, user else user # Checks if user is just id for presence_change
 
     # Send to Hubot based on message type
     if event.type is 'message'
-      
-      # NOTE: use robot.brain.userForId(id, options) to initialize the user object
-      # think about whether this is true for bots and if we want to be storing bots in the same brain namespace as users
-      # Hubot expects this format for TextMessage Listener
+      # If bot user not found in users.info from client
```

Comment:
i don't understand this comment. would it be accurate to say "If no user in message, set a default value of an empty object"? and if so, isn't this handled by the `@robot.brain.userForId()` method? idk, i'm confused.

Reply:
That's a better way to state it, yeah. The `userForId()` method doesn't handle this because it requires a non-empty ID, which would not exist in this scenario.